File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
// 올바른 괄호
// (, ) 로만 구성된 문자열

// s: 1~10만
// O(N^2)까지 가능

import Foundation

func solution(_ s:String) -> Bool {
    
    var stack: [Character] = []
    
    for c in s {
        if c == "(" {
            stack.append(c)
        } else {
            if stack.isEmpty { return false } 
            else { stack.popLast() }
        }
    }

    return stack.isEmpty
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
# 괄호 확인
# stack > 10만개의 배열 체크

def solution(s):
    
    stack = []
    for st in s:
        if len(stack) > 0 and st == ')' and stack[-1] == '(':
            stack.pop()
        else:
            stack.append(st)
        
    return len(stack) == 0